File: aulas-2016-17/FormApp/app/src/main/java/com/wolneyhqf/formapp/ListaActivity.java
package com.wolneyhqf.formapp;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import com.wolneyhqf.formapp.adaptadores.AlunosAdapter;
import com.wolneyhqf.formapp.modelo.Aluno;

import java.util.ArrayList;
import java.util.List;

public class ListaActivity extends AppCompatActivity {

    private ListView listView;
    private List<Aluno> alunos;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_lista);

        listView = (ListView) findViewById(R.id.listview_alunos);
        final List<Aluno> alunos = getAlunos();
        AlunosAdapter adaptador = new AlunosAdapter(alunos, this);

        listView.setAdapter(adaptador);

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                Aluno aluno = alunos.get(i);
                Intent intent = new Intent(ListaActivity.this, AlunoActivity.class);
                intent.putExtra("aluno", aluno);
                startActivity(intent);
            }
        });
    }

    private List<Aluno> getAlunos(){
        List<Aluno> alunos = new ArrayList<>();
        alunos.add(new Aluno("José", "1001", "josej@gmail.com"));
        alunos.add(new Aluno("Joana", "1002", "joana@gmail.com"));
        alunos.add(new Aluno("Alberto", "1003", "alberto@gmail.com"));
        alunos.add(new Aluno("Pedro", "1004", "pedro1444@gmail.com"));
        return alunos;
    }

}

File: 2018/Aula1711/app/src/main/java/com/eecm/aula0911/ListaActivity.kt
package com.eecm.aula0911

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast

class ListaActivity : AppCompatActivity() {

    private var clientes = listOf<Cliente>()
    var recyclerView: RecyclerView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_lista)

        supportActionBar?.title = "Lista"
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
        recyclerView?.layoutManager = LinearLayoutManager(this)
        recyclerView?.itemAnimator = DefaultItemAnimator()
        recyclerView?.setHasFixedSize(true)

        taskClientes()
    }

    fun taskClientes(){
        this.clientes = ClienteService.getClientes()
        recyclerView?.adapter = ClienteAdapter(clientes,
            {
                cliente: Cliente -> Toast.makeText(this, "@Clicou no cliente ${cliente.nome}", Toast.LENGTH_SHORT).show()
            }
        )
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        var id = item?.itemId
        if(id == android.R.id.home){
            var it = Intent(this, MainActivity::class.java)
            startActivity(it)
        }else if(id == R.id.action_search){
            Toast.makeText(this, "Search", Toast.LENGTH_SHORT).show()
        }
        return super.onOptionsItemSelected(item)
    }

}
